ast-nodes/statement.dart

```diff
@@ -1,9 +1,36 @@
+import '../syntax-error.dart';
 import 'node.dart';
 import '../lexer.dart';
+import 'while-loop.dart';
+import 'for-loop.dart';
+import 'if-statement.dart';
+import 'routine-call.dart';
+import 'assignment.dart';
 
 /// An abstract statement.
 abstract class Statement implements Node {
   factory Statement.parse(Iterable<Token> tokens) {
+    if (tokens.isEmpty) {
+      throw SyntaxError(tokens?.first, 'Expected a statement');
```

Comment:
Same thing about redundant optional chaining, in the condition of the `if` you've used _unprotected_ access anyway, so even if `tokens` was `null`, you'd get an error.